qutebrowser/mainwindow/statusbar/command.py

```diff
@@ -114,7 +119,11 @@ def set_cmd_text_command(self, text, space=False, append=False):
         if not text or text[0] not in modeparsers.STARTCHARS:
             raise cmdexc.CommandError(
                 "Invalid command text '{}'.".format(text))
-        self.set_cmd_text(text)
+        if run_on_count and count is not None:
+            self.set_cmd_text(':run-with-count {} {}'.format(count, text))
+            self.command_accept()
```

Comment:
That seems kind of a hack - instead, you could change the `got_cmd` signal to `pyqtSignal([str], [str, int])` and then do `self.got_cmd.emit(text, count)`. I think that should work, as `got_cmd` is connect to `CommandRunner.run_safely` which understands both forms already.